### PR TITLE
New DocIndex

### PR DESCRIPTION
**Pull request by JimSuplizio**
Add the code to create the new DocIndex. An example of what this looks like can be found [here](https://jimsuplizio.github.io/azure-sdk-for-java/). This new DocIndex handles versioned docs. The doc, themselves, are stored in azure blob storage (I'm still doing the initial population so only the index is there right now, it'll be about 30 minutes or so). 

It's also worth noting that while there are changes to pom.client.xml and pom.data.xml these changes will NOT cause us to have to republish them. The reason for this change was to pipe the version into the script that generates the additional overview from the readme so we can have the version on the landing page with a link back to the index.

### CODE REVIEW

File: eng/pipelines/docindex.yml [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+          overWrite: true
+      - pwsh: ls
+        workingDirectory: $(Build.SourcesDirectory)/docfx_project/
+      - pwsh: ls
+        workingDirectory: $(Build.SourcesDirectory)/docfx_project/templates
+      - pwsh: |
+          Copy-Item "$(Build.SourcesDirectory)/eng/docgeneration/docfx.json" -Destination "$(Build.SourcesDirectory)/docfx_project/" -Force
+        displayName: Copy over docfx.json
+      - pwsh: |
+
```
Comment (scbedd) [inline eng/pipelines/docindex.yml]: I feel like this should be an external powershell hosted in the `eng/` directory or the like. Maybe `eng/docgeneration`. 

However, I think that's the case for this entire thing, and both `JS` and `.NET` have the same problem. So I don't want to block this checkin based on this minor issue.

Comment (JimSuplizio) [reply eng/pipelines/docindex.yml]: @scbedd That's a fair enough request. I'll rework this when you file the issue.

**Comment by JimSuplizio** [comment]
Because the azp run commands weren't working here I'd ended up kicking off a run through the pipeline, itself. Since that passed I'm going to check this in.

**Pull request merged**